In [44]:
import numpy as np
import pandas as pd
import matplotlib as mlt
import seaborn as sns
import matplotlib.pyplot as plt

In [45]:
df = pd.read_csv(r"C:\Users\prash\Downloads\ETL (1)\ETL (1)\ETL (1)\DATA\100 Sales Records.csv")

In [46]:
df

Region                Country        Item Type  \
0               Australia and Oceania                 Tuvalu        Baby Food   
1   Central America and the Caribbean                Grenada           Cereal   
2                              Europe                 Russia  Office Supplies   
3                  Sub-Saharan Africa  Sao Tome and Principe           Fruits   
4                  Sub-Saharan Africa                 Rwanda  Office Supplies   
..                                ...                    ...              ...   
95                 Sub-Saharan Africa                   Mali          Clothes   
96                               Asia               Malaysia           Fruits   
97                 Sub-Saharan Africa           Sierra Leone       Vegetables   
98                      North America                 Mexico    Personal Care   
99                 Sub-Saharan Africa             Mozambique        Household   

   Sales Channel Order Priority  Order Date   Order ID   Ship Date  \
0        Offline              H   5/28/2010  669165933   6/27/2010   
1         Online              C   8/22/2012  963881480   9/15/2012   
2        Offline              L  05-02-2014  341417157  05-08-2014   
3         Online              C   6/20/2014  514321792  07-05-2014   
4        Offline              L  02-01-2013  115456712  02-06-2013   
..           ...            ...         ...        ...         ...   
95        Online              M   7/26/2011  512878119  09-03-2011   
96       Offline              L  11-11-2011  810711038  12/28/2011   
97       Offline              C  06-01-2016  728815257   6/29/2016   
98       Offline              M   7/30/2015  559427106  08-08-2015   
99       Offline              L  02-10-2012  665095412   2/15/2012   

    Units Sold  Unit Price  Unit Cost  Total Revenue  Total Cost  Total Profit  
0         9925      255.28     159.42     2533654.00  1582243.50     951410.50  
1         2804      205.70     117.11      576782.80   328376.44     248406.36  
2         1779      651.21     524.96     1158502.59   933903.84     224598.75  
3         8102        9.33       6.92       75591.66    56065.84      19525.82  
4         5062      651.21     524.96     3296425.02  2657347.52     639077.50  
..         ...         ...        ...            ...         ...           ...  
95         888      109.28      35.84       97040.64    31825.92      65214.72  
96        6267        9.33       6.92       58471.11    43367.64      15103.47  
97        1485      154.06      90.93      228779.10   135031.05      93748.05  
98        5767       81.73      56.67      471336.91   326815.89     144521.02  
99        5367      668.27     502.54     3586605.09  2697132.18     889472.91  

[100 rows x 14 columns]

In [4]:
df.nunique()

Region              7
Country            76
Item Type          12
Sales Channel       2
Order Priority      4
Order Date        100
Order ID          100
Ship Date          99
Units Sold         99
Unit Price         12
Unit Cost          12
Total Revenue     100
Total Cost        100
Total Profit      100
dtype: int64

In [5]:
df.describe()

Order ID   Units Sold  Unit Price   Unit Cost  Total Revenue  \
count  1.000000e+02   100.000000  100.000000  100.000000   1.000000e+02   
mean   5.550204e+08  5128.710000  276.761300  191.048000   1.373488e+06   
std    2.606153e+08  2794.484562  235.592241  188.208181   1.460029e+06   
min    1.146066e+08   124.000000    9.330000    6.920000   4.870260e+03   
25%    3.389225e+08  2836.250000   81.730000   35.840000   2.687212e+05   
50%    5.577086e+08  5382.500000  179.880000  107.275000   7.523144e+05   
75%    7.907551e+08  7369.000000  437.200000  263.330000   2.212045e+06   
max    9.940222e+08  9925.000000  668.270000  524.960000   5.997055e+06   

         Total Cost  Total Profit  
count  1.000000e+02  1.000000e+02  
mean   9.318057e+05  4.416820e+05  
std    1.083938e+06  4.385379e+05  
min    3.612240e+03  1.258020e+03  
25%    1.688680e+05  1.214436e+05  
50%    3.635664e+05  2.907680e+05  
75%    1.613870e+06  6.358288e+05  
max    4.509794e+06  1.719922e+06

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 14 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Region          100 non-null    object 
 1   Country         100 non-null    object 
 2   Item Type       100 non-null    object 
 3   Sales Channel   100 non-null    object 
 4   Order Priority  100 non-null    object 
 5   Order Date      100 non-null    object 
 6   Order ID        100 non-null    int64  
 7   Ship Date       100 non-null    object 
 8   Units Sold      100 non-null    int64  
 9   Unit Price      100 non-null    float64
 10  Unit Cost       100 non-null    float64
 11  Total Revenue   100 non-null    float64
 12  Total Cost      100 non-null    float64
 13  Total Profit    100 non-null    float64
dtypes: float64(5), int64(2), object(7)
memory usage: 11.1+ KB


In [7]:
df.columns

Index(['Region', 'Country', 'Item Type', 'Sales Channel', 'Order Priority',
       'Order Date', 'Order ID', 'Ship Date', 'Units Sold', 'Unit Price',
       'Unit Cost', 'Total Revenue', 'Total Cost', 'Total Profit'],
      dtype='object')

In [8]:
df.shape 
# 14 Columns
# 100 Rows

(100, 14)

In [9]:
df["Region"].unique()

array(['Australia and Oceania', 'Central America and the Caribbean',
       'Europe', 'Sub-Saharan Africa', 'Asia',
       'Middle East and North Africa', 'North America'], dtype=object)

In [10]:
df["Region"].nunique()

7

In [11]:
df["Region"].value_counts()

Sub-Saharan Africa                   36
Europe                               22
Australia and Oceania                11
Asia                                 11
Middle East and North Africa         10
Central America and the Caribbean     7
North America                         3
Name: Region, dtype: int64

**Maximum orders are from Sub-Saharan Africa Region. Total= 36 orders.**

In [12]:
df[(df["Region"]=="Sub-Saharan Africa") & (df["Sales Channel"]=="Offline")]

Region                Country        Item Type Sales Channel  \
4   Sub-Saharan Africa                 Rwanda  Office Supplies       Offline   
6   Sub-Saharan Africa                 Angola        Household       Offline   
8   Sub-Saharan Africa  Republic of the Congo    Personal Care       Offline   
11  Sub-Saharan Africa             Cape Verde          Clothes       Offline   
17  Sub-Saharan Africa               Cameroon        Beverages       Offline   
29  Sub-Saharan Africa             The Gambia        Household       Offline   
31  Sub-Saharan Africa            South Sudan    Personal Care       Offline   
42  Sub-Saharan Africa             The Gambia           Cereal       Offline   
45  Sub-Saharan Africa                Comoros           Cereal       Offline   
49  Sub-Saharan Africa             Mauritania  Office Supplies       Offline   
53  Sub-Saharan Africa           Sierra Leone  Office Supplies       Offline   
54  Sub-Saharan Africa  Sao Tome and Principe           Fruits       Offline   
59  Sub-Saharan Africa               Djibouti        Cosmetics       Offline   
65  Sub-Saharan Africa                 Rwanda        Cosmetics       Offline   
66  Sub-Saharan Africa                  Gabon    Personal Care       Offline   
69  Sub-Saharan Africa             Madagascar          Clothes       Offline   
86  Sub-Saharan Africa  Sao Tome and Principe        Beverages       Offline   
87  Sub-Saharan Africa             The Gambia        Baby Food       Offline   
90  Sub-Saharan Africa           Sierra Leone  Office Supplies       Offline   
97  Sub-Saharan Africa           Sierra Leone       Vegetables       Offline   
99  Sub-Saharan Africa             Mozambique        Household       Offline   

   Order Priority  Order Date   Order ID   Ship Date  Units Sold  Unit Price  \
4               L  02-01-2013  115456712  02-06-2013        5062      651.21   
6               M   4/23/2011  135425221   4/27/2011        4187      668.27   
8               M   7/14/2015  770463311   8/25/2015        6070       81.73   
11              H  08-02-2014  939825713   8/19/2014        4168      109.28   
17              C  04-01-2015  519820964   4/18/2015        5430       47.45   
29              L   5/26/2012  886494815  06-09-2012        2370      668.27   
31              C  12/29/2013  406502997   1/28/2014        2125       81.73   
42              H  06-07-2012  994022214  06-08-2012        2117      205.70   
45              H   3/29/2016  902102267   4/29/2016         962      205.70   
49              C  01-11-2012  837559306   1/13/2012        1266      651.21   
53              M  11/26/2011  441888415  01-07-2012        3457      651.21   
54              H   9/17/2013  508980977  10/24/2013        7637        9.33   
59              H  04-07-2014  259353148   4/19/2014        7215      437.20   
65              H  10-11-2013  699358165  11/25/2013        4477      437.20   
66              L  07-08-2012  228944623  07-09-2012        8656       81.73   
69              L   4/25/2015  610425555   5/28/2015        7342      109.28   
86              C   1/16/2011  180283772   1/21/2011        8829       47.45   
87              M  02-03-2014  494747245   3/20/2014        5559      255.28   
90              H  12-06-2016  621386563  12/14/2016         948      651.21   
97              C  06-01-2016  728815257   6/29/2016        1485      154.06   
99              L  02-10-2012  665095412   2/15/2012        5367      668.27   

    Unit Cost  Total Revenue  Total Cost  Total Profit  
4      524.96     3296425.02  2657347.52     639077.50  
6      502.54     2798046.49  2104134.98     693911.51  
8       56.67      496101.10   343986.90     152114.20  
11      35.84      455479.04   149381.12     306097.92  
17      31.79      257653.50   172619.70      85033.80  
29     502.54     1583799.90  1191019.80     392780.10  
31      56.67      173676.25   120423.75      53252.50  
42     117.11      435466.90   247921.

 **21/36** orders from _"Sub-Saharan Africa"_ are **online**.
 
 **15/36** are **offline**.

In [13]:
df[(df["Region"]=="Sub-Saharan Africa") & (df["Sales Channel"]=="Offline")].nunique()

Region             1
Country           15
Item Type         10
Sales Channel      1
Order Priority     4
Order Date        21
Order ID          21
Ship Date         21
Units Sold        21
Unit Price        10
Unit Cost         10
Total Revenue     21
Total Cost        21
Total Profit      21
dtype: int64

In [14]:
# Shape of the data set
df.shape

(100, 14)

In [15]:
# Country unique names.
df["Country"].unique()

array(['Tuvalu', 'Grenada', 'Russia', 'Sao Tome and Principe', 'Rwanda',
       'Solomon Islands', 'Angola', 'Burkina Faso',
       'Republic of the Congo', 'Senegal', 'Kyrgyzstan', 'Cape Verde',
       'Bangladesh', 'Honduras', 'Mongolia', 'Bulgaria', 'Sri Lanka',
       'Cameroon', 'Turkmenistan', 'East Timor', 'Norway', 'Portugal',
       'New Zealand', 'Moldova ', 'France', 'Kiribati', 'Mali',
       'The Gambia', 'Switzerland', 'South Sudan', 'Australia', 'Myanmar',
       'Djibouti', 'Costa Rica', 'Syria', 'Brunei', 'Niger', 'Azerbaijan',
       'Slovakia', 'Comoros', 'Iceland', 'Macedonia', 'Mauritania',
       'Albania', 'Lesotho', 'Saudi Arabia', 'Sierra Leone',
       "Cote d'Ivoire", 'Fiji', 'Austria', 'United Kingdom', 'San Marino',
       'Libya', 'Haiti', 'Gabon', 'Belize', 'Lithuania', 'Madagascar',
       'Democratic Republic of the Congo', 'Pakistan', 'Mexico',
       'Federated States of Micronesia', 'Laos', 'Monaco', 'Samoa ',
       'Spain', 'Lebanon', 'Iran', 'Zamb

In [16]:
# Total countries.
df["Country"].nunique() 

76

In [17]:
# item types
df["Item Type"]

0           Baby Food
1              Cereal
2     Office Supplies
3              Fruits
4     Office Supplies
           ...       
95            Clothes
96             Fruits
97         Vegetables
98      Personal Care
99          Household
Name: Item Type, Length: 100, dtype: object

In [18]:
df["Item Type"].unique()

array(['Baby Food', 'Cereal', 'Office Supplies', 'Fruits', 'Household',
       'Vegetables', 'Personal Care', 'Clothes', 'Cosmetics', 'Beverages',
       'Meat', 'Snacks'], dtype=object)

In [19]:
df["Item Type"].nunique() 


12

 There are 12 different types of items. 
 - Most items belong to the clothes and cosmetics category. 13 each. Total count = 26/100
 - Least items belong to vegetables category. Only 6.

In [20]:
df["Sales Channel"].unique()

array(['Offline', 'Online'], dtype=object)

In [21]:
df["Sales Channel"].value_counts()

Online     50
Offline    50
Name: Sales Channel, dtype: int64

 Total sales made online and offline are same. ie. 50 - 50 out of 100.

In [22]:
df["Order Priority"].unique()

array(['H', 'C', 'L', 'M'], dtype=object)

- Critical,
- High,  
- Medium, 
- Low

In [23]:
df["Order Priority"].value_counts()

H    30
L    27
C    22
M    21
Name: Order Priority, dtype: int64

In [24]:
df.info

<bound method DataFrame.info of                                Region                Country        Item Type  \
0               Australia and Oceania                 Tuvalu        Baby Food   
1   Central America and the Caribbean                Grenada           Cereal   
2                              Europe                 Russia  Office Supplies   
3                  Sub-Saharan Africa  Sao Tome and Principe           Fruits   
4                  Sub-Saharan Africa                 Rwanda  Office Supplies   
..                                ...                    ...              ...   
95                 Sub-Saharan Africa                   Mali          Clothes   
96                               Asia               Malaysia           Fruits   
97                 Sub-Saharan Africa           Sierra Leone       Vegetables   
98                      North America                 Mexico    Personal Care   
99                 Sub-Saharan Africa             Mozambique        Household

In [25]:
df.info(max_cols=len(df))


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 14 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Region          100 non-null    object 
 1   Country         100 non-null    object 
 2   Item Type       100 non-null    object 
 3   Sales Channel   100 non-null    object 
 4   Order Priority  100 non-null    object 
 5   Order Date      100 non-null    object 
 6   Order ID        100 non-null    int64  
 7   Ship Date       100 non-null    object 
 8   Units Sold      100 non-null    int64  
 9   Unit Price      100 non-null    float64
 10  Unit Cost       100 non-null    float64
 11  Total Revenue   100 non-null    float64
 12  Total Cost      100 non-null    float64
 13  Total Profit    100 non-null    float64
dtypes: float64(5), int64(2), object(7)
memory usage: 11.1+ KB


In [26]:
df["Order Date"]

0      5/28/2010
1      8/22/2012
2     05-02-2014
3      6/20/2014
4     02-01-2013
         ...    
95     7/26/2011
96    11-11-2011
97    06-01-2016
98     7/30/2015
99    02-10-2012
Name: Order Date, Length: 100, dtype: object

In [27]:
df["Order ID"].nunique()


100

 All customers are unique/different.

In [28]:
df["Ship Date"]

0      6/27/2010
1      9/15/2012
2     05-08-2014
3     07-05-2014
4     02-06-2013
         ...    
95    09-03-2011
96    12/28/2011
97     6/29/2016
98    08-08-2015
99     2/15/2012
Name: Ship Date, Length: 100, dtype: object

In [29]:
df["Units Sold"]

0     9925
1     2804
2     1779
3     8102
4     5062
      ... 
95     888
96    6267
97    1485
98    5767
99    5367
Name: Units Sold, Length: 100, dtype: int64

In [30]:
# Sorting values by maximum "Units sold" , in descending order.
df.sort_values("Units Sold", ascending = False)

Region         Country      Item Type Sales Channel  \
0          Australia and Oceania          Tuvalu      Baby Food       Offline   
56         Australia and Oceania            Fiji        Clothes       Offline   
74  Middle East and North Africa        Pakistan      Cosmetics       Offline   
79         Australia and Oceania          Samoa       Cosmetics        Online   
51            Sub-Saharan Africa         Lesotho         Fruits        Online   
..                           ...             ...            ...           ...   
88  Middle East and North Africa          Kuwait         Fruits        Online   
58                        Europe  United Kingdom      Household        Online   
47                        Europe     Switzerland  Personal Care        Online   
43                        Europe        Slovakia     Vegetables        Online   
10                          Asia      Kyrgyzstan     Vegetables        Online   

   Order Priority  Order Date   Order ID   Ship Date  Units Sold  Unit Price  \
0               H   5/28/2010  669165933   6/27/2010        9925      255.28   
56              C   6/30/2010  647876489  08-01-2010        9905      109.28   
74              L  07-05-2013  231145322   8/16/2013        9892      437.20   
79              H   7/20/2013  670854651  08-07-2013        9654      437.20   
51              L   8/18/2013  918419539   9/18/2013        9606        9.33   
..            ...         ...        ...         ...         ...         ...   
88              M   4/30/2012  513417565   5/18/2012         522        9.33   
58              L  01-05-2012  955357205   2/14/2012         282      668.27   
47              M  12/23/2010  617667090   1/31/2011         273       81.73   
43              H  10-06-2012  759224212  11-10-2012         171      154.06   
10              H   6/24/2011  814711606  07-12-2011         124      154.06   

    Unit Cost  Total Revenue  Total Cost  Total Profit  
0      159.42     2533654.00  1582243.50     951410.50  
56      35.84     1082418.40   354995.20     727423.20  
74     263.33     4324782.40  2604860.36    1719922.04  
79     263.33     4220728.80  2542187.82    1678540.98  
51       6.92       89623.98    66473.52      23150.46  
..        ...            ...         ...           ...  
88       6.92        4870.26     3612.24       1258.02  
58     502.54      188452.14   141716.28      46735.86  
47      56.67       22312.29    15470.91       6841.38  
43      90.93       26344.26    15549.03      10795.23  
10      90.93       19103.44    11275.32       7828.12  

[100 rows x 14 columns]

In [31]:
df.sort_values("Unit Price", ascending = False).head(10)

Region         Country        Item Type  \
99                 Sub-Saharan Africa      Mozambique        Household   
33                               Asia         Myanmar        Household   
6                  Sub-Saharan Africa          Angola        Household   
29                 Sub-Saharan Africa      The Gambia        Household   
58                             Europe  United Kingdom        Household   
80                             Europe           Spain        Household   
13  Central America and the Caribbean        Honduras        Household   
75                      North America          Mexico        Household   
18                               Asia    Turkmenistan        Household   
70                               Asia    Turkmenistan  Office Supplies   

   Sales Channel Order Priority  Order Date   Order ID   Ship Date  \
99       Offline              L  02-10-2012  665095412   2/15/2012   
33       Offline              H   1/16/2015  177713572  03-01-2015   
6        Offline              M   4/23/2011  135425221   4/27/2011   
29       Offline              L   5/26/2012  886494815  06-09-2012   
58        Online              L  01-05-2012  955357205   2/14/2012   
80       Offline              L  10/21/2012  213487374  11/30/2012   
13       Offline              H  02-08-2017  522840487   2/13/2017   
75       Offline              C  11-06-2014  986435210  12-12-2014   
18       Offline              L  12/30/2010  441619336   1/20/2011   
70        Online              M   4/23/2013  462405812   5/20/2013   

    Units Sold  Unit Price  Unit Cost  Total Revenue  Total Cost  Total Profit  
99        5367      668.27     502.54     3586605.09  2697132.18     889472.91  
33        8250      668.27     502.54     5513227.50  4145955.00    1367272.50  
6         4187      668.27     502.54     2798046.49  2104134.98     693911.51  
29        2370      668.27     502.54     1583799.90  1191019.80     392780.10  
58         282      668.27     502.54      188452.14   141716.28      46735.86  
80        4513      668.27     502.54     3015902.51  2267963.02     747939.49  
13        8974      668.27     502.54     5997054.98  4509793.96    1487261.02  
75        6954      668.27     502.54     4647149.58  3494663.16    1152486.42  
18        3830      668.27     502.54     2559474.10  1924728.20     634745.90  
70        5010      651.21     524.96     3262562.10  2630049.60     632512.50

Household is the costliest item type, with maximum unit cost 668.27. But sold at 502.54.
- 9 regions bought total.

In [32]:
df.sort_values("Unit Price", ascending = False).tail(15)

Region                           Country  Item Type  \
91         Australia and Oceania                         Australia  Beverages   
89                        Europe                          Slovenia  Beverages   
17            Sub-Saharan Africa                          Cameroon  Beverages   
86            Sub-Saharan Africa             Sao Tome and Principe  Beverages   
72            Sub-Saharan Africa  Democratic Republic of the Congo  Beverages   
27            Sub-Saharan Africa                              Mali     Fruits   
54            Sub-Saharan Africa             Sao Tome and Principe     Fruits   
26         Australia and Oceania                          Kiribati     Fruits   
23         Australia and Oceania                       New Zealand     Fruits   
3             Sub-Saharan Africa             Sao Tome and Principe     Fruits   
36  Middle East and North Africa                             Syria     Fruits   
51            Sub-Saharan Africa                           Lesotho     Fruits   
96                          Asia                          Malaysia     Fruits   
71  Middle East and North Africa                             Libya     Fruits   
88  Middle East and North Africa                            Kuwait     Fruits   

   Sales Channel Order Priority  Order Date   Order ID   Ship Date  \
91       Offline              H  07-07-2014  240470397  07-11-2014   
89       Offline              C  10/23/2016  345718562  11/25/2016   
17       Offline              C  04-01-2015  519820964   4/18/2015   
86       Offline              C   1/16/2011  180283772   1/21/2011   
72        Online              C   5/26/2011  585920464   7/15/2011   
27        Online              L  05-07-2010  686048400  05-10-2010   
54       Offline              H   9/17/2013  508980977  10/24/2013   
26        Online              M  10/13/2014  347140347  11-10-2014   
23        Online              H  09-08-2014  142278373  10-04-2014   
3         Online              C   6/20/2014  514321792  07-05-2014   
36        Online              L  11/22/2011  162052476  12-03-2011   
51        Online              L   8/18/2013  918419539   9/18/2013   
96       Offline              L  11-11-2011  810711038  12/28/2011   
71        Online              L   8/14/2015  816200339   9/30/2015   
88        Online              M   4/30/2012  513417565   5/18/2012   

    Units Sold  Unit Price  Unit Cost  Total Revenue  Total Cost  Total Profit  
91        9389       47.45      31.79      445508.05   298476.31     147031.74  
89        4660       47.45      31.79      221117.00   148141.40      72975.60  
17        5430       47.45      31.79      257653.50   172619.70      85033.80  
86        8829       47.45      31.79      418936.05   280673.91     138262.14  
72        5741       47.45      31.79      272410.45   182506.39      89904.06  
27        5822        9.33       6.92       54319.26    40288.24      14031.02  
54        7637        9.33       6.92       71253.21    52848.04      18405.17  
26        5398        9.33       6.92       50363.34    37354.16      13009.18  
23        2187        9.33       6.92       20404.71    15134.04       5270.67  
3         8102        9.33       6.92       75591.66    56065.84      19525.82  
36        3784        9.33       6.92       35304.72    26185.28       9119.44  
51        9606        9.33       6.92       89623.98    66473.52      23150.46  
96        6267        9.33       6.92       58471.11    43367.64      15103.47  
71         673        9.33       6.92        6279.09     4657.16       1621.93  
88         522        9.33       6.92        4870.26     3612.24       1258.02

*Fruits* is the cheapest item type, with
- minimum unit cost 9.33 . But sold at 6.92.

In [33]:
df.sort_values("Unit Cost", ascending = False).head(15)

# Please refer the OCCURENCE graphs generated by LUX.

Region         Country        Item Type  \
38                          Asia          Brunei  Office Supplies   
70                          Asia    Turkmenistan  Office Supplies   
2                         Europe          Russia  Office Supplies   
49            Sub-Saharan Africa      Mauritania  Office Supplies   
4             Sub-Saharan Africa          Rwanda  Office Supplies   
92  Middle East and North Africa      Azerbaijan  Office Supplies   
90            Sub-Saharan Africa    Sierra Leone  Office Supplies   
32         Australia and Oceania       Australia  Office Supplies   
53            Sub-Saharan Africa    Sierra Leone  Office Supplies   
62            Sub-Saharan Africa        Cameroon  Office Supplies   
68                        Europe       Lithuania  Office Supplies   
39                        Europe        Bulgaria  Office Supplies   
33                          Asia         Myanmar        Household   
29            Sub-Saharan Africa      The Gambia        Household   
58                        Europe  United Kingdom        Household   

   Sales Channel Order Priority  Order Date   Order ID   Ship Date  \
38        Online              L  04-01-2012  320009267  05-08-2012   
70        Online              M   4/23/2013  462405812   5/20/2013   
2        Offline              L  05-02-2014  341417157  05-08-2014   
49       Offline              C  01-11-2012  837559306   1/13/2012   
4        Offline              L  02-01-2013  115456712  02-06-2013   
92        Online              M   6/13/2012  423331391   7/24/2012   
90       Offline              H  12-06-2016  621386563  12/14/2016   
32        Online              C  10/27/2015  158535134  11/25/2015   
53       Offline              M  11/26/2011  441888415  01-07-2012   
62        Online              M  11-07-2011  177636754  11/15/2011   
68       Offline              H  10/24/2010  166460740  11/17/2010   
39        Online              M   2/16/2012  189965903   2/28/2012   
33       Offline              H   1/16/2015  177713572  03-01-2015   
29       Offline              L   5/26/2012  886494815  06-09-2012   
58        Online              L  01-05-2012  955357205   2/14/2012   

    Units Sold  Unit Price  Unit Cost  Total Revenue  Total Cost  Total Profit  
38        6708      651.21     524.96     4368316.68  3521431.68     846885.00  
70        5010      651.21     524.96     3262562.10  2630049.60     632512.50  
2         1779      651.21     524.96     1158502.59   933903.84     224598.75  
49        1266      651.21     524.96      824431.86   664599.36     159832.50  
4         5062      651.21     524.96     3296425.02  2657347.52     639077.50  
92        2021      651.21     524.96     1316095.41  1060944.16     255151.25  
90         948      651.21     524.96      617347.08   497662.08     119685.00  
32        2924      651.21     524.96     1904138.04  1534983.04     369155.00  
53        3457      651.21     524.96     2251232.97  1814786.72     436446.25  
62        5518      651.21     524.96     3593376.78  2896729.28     696647.50  
68        8287      651.21     524.96     5396577.27  4350343.52    1046233.75  
39        3987      651.21     524.96     2596374.27  2093015.52     503358.75  
33        8250      668.27     502.54     5513227.50  4145955.00    1367272.50  
29        2370      668.27     502.54     1583799.90  1191019.80     392780.10  
58         282      668.27     502.54      188452.14   141716.28      46735.86

*Office supplies*

    - 2nd highest unit price
    - highest unit cost
    - 12 regions bought(total)

In [34]:
# Descending sorting of Total Revenue generated.

df.sort_values("Total Revenue", ascending = False)

Region      Country        Item Type  \
13  Central America and the Caribbean     Honduras        Household   
33                               Asia      Myanmar        Household   
68                             Europe    Lithuania  Office Supplies   
75                      North America       Mexico        Household   
38                               Asia       Brunei  Office Supplies   
..                                ...          ...              ...   
47                             Europe  Switzerland    Personal Care   
23              Australia and Oceania  New Zealand           Fruits   
10                               Asia   Kyrgyzstan       Vegetables   
71       Middle East and North Africa        Libya           Fruits   
88       Middle East and North Africa       Kuwait           Fruits   

   Sales Channel Order Priority  Order Date   Order ID   Ship Date  \
13       Offline              H  02-08-2017  522840487   2/13/2017   
33       Offline              H   1/16/2015  177713572  03-01-2015   
68       Offline              H  10/24/2010  166460740  11/17/2010   
75       Offline              C  11-06-2014  986435210  12-12-2014   
38        Online              L  04-01-2012  320009267  05-08-2012   
..           ...            ...         ...        ...         ...   
47        Online              M  12/23/2010  617667090   1/31/2011   
23        Online              H  09-08-2014  142278373  10-04-2014   
10        Online              H   6/24/2011  814711606  07-12-2011   
71        Online              L   8/14/2015  816200339   9/30/2015   
88        Online              M   4/30/2012  513417565   5/18/2012   

    Units Sold  Unit Price  Unit Cost  Total Revenue  Total Cost  Total Profit  
13        8974      668.27     502.54     5997054.98  4509793.96    1487261.02  
33        8250      668.27     502.54     5513227.50  4145955.00    1367272.50  
68        8287      651.21     524.96     5396577.27  4350343.52    1046233.75  
75        6954      668.27     502.54     4647149.58  3494663.16    1152486.42  
38        6708      651.21     524.96     4368316.68  3521431.68     846885.00  
..         ...         ...        ...            ...         ...           ...  
47         273       81.73      56.67       22312.29    15470.91       6841.38  
23        2187        9.33       6.92       20404.71    15134.04       5270.67  
10         124      154.06      90.93       19103.44    11275.32       7828.12  
71         673        9.33       6.92        6279.09     4657.16       1621.93  
88         522        9.33       6.92        4870.26     3612.24       1258.02  

[100 rows x 14 columns]

In [35]:
# Descending sorting of Total Cost generated.

df.sort_values("Total Cost", ascending = False)

Region      Country        Item Type  \
13  Central America and the Caribbean     Honduras        Household   
68                             Europe    Lithuania  Office Supplies   
33                               Asia      Myanmar        Household   
38                               Asia       Brunei  Office Supplies   
75                      North America       Mexico        Household   
..                                ...          ...              ...   
47                             Europe  Switzerland    Personal Care   
23              Australia and Oceania  New Zealand           Fruits   
10                               Asia   Kyrgyzstan       Vegetables   
71       Middle East and North Africa        Libya           Fruits   
88       Middle East and North Africa       Kuwait           Fruits   

   Sales Channel Order Priority  Order Date   Order ID   Ship Date  \
13       Offline              H  02-08-2017  522840487   2/13/2017   
68       Offline              H  10/24/2010  166460740  11/17/2010   
33       Offline              H   1/16/2015  177713572  03-01-2015   
38        Online              L  04-01-2012  320009267  05-08-2012   
75       Offline              C  11-06-2014  986435210  12-12-2014   
..           ...            ...         ...        ...         ...   
47        Online              M  12/23/2010  617667090   1/31/2011   
23        Online              H  09-08-2014  142278373  10-04-2014   
10        Online              H   6/24/2011  814711606  07-12-2011   
71        Online              L   8/14/2015  816200339   9/30/2015   
88        Online              M   4/30/2012  513417565   5/18/2012   

    Units Sold  Unit Price  Unit Cost  Total Revenue  Total Cost  Total Profit  
13        8974      668.27     502.54     5997054.98  4509793.96    1487261.02  
68        8287      651.21     524.96     5396577.27  4350343.52    1046233.75  
33        8250      668.27     502.54     5513227.50  4145955.00    1367272.50  
38        6708      651.21     524.96     4368316.68  3521431.68     846885.00  
75        6954      668.27     502.54     4647149.58  3494663.16    1152486.42  
..         ...         ...        ...            ...         ...           ...  
47         273       81.73      56.67       22312.29    15470.91       6841.38  
23        2187        9.33       6.92       20404.71    15134.04       5270.67  
10         124      154.06      90.93       19103.44    11275.32       7828.12  
71         673        9.33       6.92        6279.09     4657.16       1621.93  
88         522        9.33       6.92        4870.26     3612.24       1258.02  

[100 rows x 14 columns]

In [36]:
# Descending sorting of Total Profit generated.

df.sort_values("Total Profit", ascending = False)

Region      Country      Item Type  \
74       Middle East and North Africa     Pakistan      Cosmetics   
79              Australia and Oceania       Samoa       Cosmetics   
46                             Europe      Iceland      Cosmetics   
30                             Europe  Switzerland      Cosmetics   
13  Central America and the Caribbean     Honduras      Household   
..                                ...          ...            ...   
10                               Asia   Kyrgyzstan     Vegetables   
47                             Europe  Switzerland  Personal Care   
23              Australia and Oceania  New Zealand         Fruits   
71       Middle East and North Africa        Libya         Fruits   
88       Middle East and North Africa       Kuwait         Fruits   

   Sales Channel Order Priority  Order Date   Order ID   Ship Date  \
74       Offline              L  07-05-2013  231145322   8/16/2013   
79        Online              H   7/20/2013  670854651  08-07-2013   
46        Online              C  12/31/2016  331438481  12/31/2016   
30       Offline              M   9/17/2012  249693334  10/20/2012   
13       Offline              H  02-08-2017  522840487   2/13/2017   
..           ...            ...         ...        ...         ...   
10        Online              H   6/24/2011  814711606  07-12-2011   
47        Online              M  12/23/2010  617667090   1/31/2011   
23        Online              H  09-08-2014  142278373  10-04-2014   
71        Online              L   8/14/2015  816200339   9/30/2015   
88        Online              M   4/30/2012  513417565   5/18/2012   

    Units Sold  Unit Price  Unit Cost  Total Revenue  Total Cost  Total Profit  
74        9892      437.20     263.33     4324782.40  2604860.36    1719922.04  
79        9654      437.20     263.33     4220728.80  2542187.82    1678540.98  
46        8867      437.20     263.33     3876652.40  2334947.11    1541705.29  
30        8661      437.20     263.33     3786589.20  2280701.13    1505888.07  
13        8974      668.27     502.54     5997054.98  4509793.96    1487261.02  
..         ...         ...        ...            ...         ...           ...  
10         124      154.06      90.93       19103.44    11275.32       7828.12  
47         273       81.73      56.67       22312.29    15470.91       6841.38  
23        2187        9.33       6.92       20404.71    15134.04       5270.67  
71         673        9.33       6.92        6279.09     4657.16       1621.93  
88         522        9.33       6.92        4870.26     3612.24       1258.02  

[100 rows x 14 columns]

- Units sold * Unit Price = Total Revenue
- Units sold * Unit Cost = Total Cost
- Total Revenue - Total Cost = Total Profit

In [37]:
df[df["Country"]=="Pakistan"]

Region   Country  Item Type Sales Channel  \
74  Middle East and North Africa  Pakistan  Cosmetics       Offline   

   Order Priority  Order Date   Order ID  Ship Date  Units Sold  Unit Price  \
74              L  07-05-2013  231145322  8/16/2013        9892       437.2   

    Unit Cost  Total Revenue  Total Cost  Total Profit  
74     263.33      4324782.4  2604860.36    1719922.04

In [38]:
df[df["Country"]=="Afghanistan"]

Empty DataFrame
Columns: [Region, Country, Item Type, Sales Channel, Order Priority, Order Date, Order ID, Ship Date, Units Sold, Unit Price, Unit Cost, Total Revenue, Total Cost, Total Profit]
Index: []

In [39]:
df.isnull().sum()

Region            0
Country           0
Item Type         0
Sales Channel     0
Order Priority    0
Order Date        0
Order ID          0
Ship Date         0
Units Sold        0
Unit Price        0
Unit Cost         0
Total Revenue     0
Total Cost        0
Total Profit      0
dtype: int64

### No null values in the dataset.
 Hence, no cleaning needed.

In [40]:
df

Region                Country        Item Type  \
0               Australia and Oceania                 Tuvalu        Baby Food   
1   Central America and the Caribbean                Grenada           Cereal   
2                              Europe                 Russia  Office Supplies   
3                  Sub-Saharan Africa  Sao Tome and Principe           Fruits   
4                  Sub-Saharan Africa                 Rwanda  Office Supplies   
..                                ...                    ...              ...   
95                 Sub-Saharan Africa                   Mali          Clothes   
96                               Asia               Malaysia           Fruits   
97                 Sub-Saharan Africa           Sierra Leone       Vegetables   
98                      North America                 Mexico    Personal Care   
99                 Sub-Saharan Africa             Mozambique        Household   

   Sales Channel Order Priority  Order Date   Order ID   Ship Date  \
0        Offline              H   5/28/2010  669165933   6/27/2010   
1         Online              C   8/22/2012  963881480   9/15/2012   
2        Offline              L  05-02-2014  341417157  05-08-2014   
3         Online              C   6/20/2014  514321792  07-05-2014   
4        Offline              L  02-01-2013  115456712  02-06-2013   
..           ...            ...         ...        ...         ...   
95        Online              M   7/26/2011  512878119  09-03-2011   
96       Offline              L  11-11-2011  810711038  12/28/2011   
97       Offline              C  06-01-2016  728815257   6/29/2016   
98       Offline              M   7/30/2015  559427106  08-08-2015   
99       Offline              L  02-10-2012  665095412   2/15/2012   

    Units Sold  Unit Price  Unit Cost  Total Revenue  Total Cost  Total Profit  
0         9925      255.28     159.42     2533654.00  1582243.50     951410.50  
1         2804      205.70     117.11      576782.80   328376.44     248406.36  
2         1779      651.21     524.96     1158502.59   933903.84     224598.75  
3         8102        9.33       6.92       75591.66    56065.84      19525.82  
4         5062      651.21     524.96     3296425.02  2657347.52     639077.50  
..         ...         ...        ...            ...         ...           ...  
95         888      109.28      35.84       97040.64    31825.92      65214.72  
96        6267        9.33       6.92       58471.11    43367.64      15103.47  
97        1485      154.06      90.93      228779.10   135031.05      93748.05  
98        5767       81.73      56.67      471336.91   326815.89     144521.02  
99        5367      668.27     502.54     3586605.09  2697132.18     889472.91  

[100 rows x 14 columns]

In [41]:
df[((df["Order Priority"]=="H") & (df["Item Type"]=="Cosmetics"))]
# choropleth maps using lux.. Mean of total profit and all..

Region   Country  Item Type Sales Channel  \
25                             Europe    France  Cosmetics        Online   
57                             Europe   Austria  Cosmetics       Offline   
59                 Sub-Saharan Africa  Djibouti  Cosmetics       Offline   
64  Central America and the Caribbean     Haiti  Cosmetics       Offline   
65                 Sub-Saharan Africa    Rwanda  Cosmetics       Offline   
79              Australia and Oceania    Samoa   Cosmetics        Online   
82       Middle East and North Africa      Iran  Cosmetics        Online   
93                             Europe   Romania  Cosmetics        Online   

   Order Priority  Order Date   Order ID   Ship Date  Units Sold  Unit Price  \
25              H   5/22/2017  898523128  06-05-2017        1815       437.2   
57              H   2/23/2015  868214595  03-02-2015        2847       437.2   
59              H  04-07-2014  259353148   4/19/2014        7215       437.2   
64              H  10/13/2013  505716836  11/16/2013        1705       437.2   
65              H  10-11-2013  699358165  11/25/2013        4477       437.2   
79              H   7/20/2013  670854651  08-07-2013        9654       437.2   
82              H  11/15/2016  286959302  12-08-2016        6489       437.2   
93              H  11/26/2010  660643374  12/25/2010        7910       437.2   

    Unit Cost  Total Revenue  Total Cost  Total Profit  
25     263.33       793518.0   477943.95     315574.05  
57     263.33      1244708.4   749700.51     495007.89  
59     263.33      3154398.0  1899925.95    1254472.05  
64     263.33       745426.0   448977.65     296448.35  
65     263.33      1957344.4  1178928.41     778415.99  
79     263.33      4220728.8  2542187.82    1678540.98  
82     263.33      2836990.8  1708748.37    1128242.43  
93     263.33      3458252.0  2082940.30    1375311.70

In [42]:
df[df["Item Type"]=="Cosmetics"].nunique()


Region             6
Country           13
Item Type          1
Sales Channel      2
Order Priority     4
Order Date        13
Order ID          13
Ship Date         13
Units Sold        13
Unit Price         1
Unit Cost          1
Total Revenue     13
Total Cost        13
Total Profit      13
dtype: int64

In [43]:
df

Region                Country        Item Type  \
0               Australia and Oceania                 Tuvalu        Baby Food   
1   Central America and the Caribbean                Grenada           Cereal   
2                              Europe                 Russia  Office Supplies   
3                  Sub-Saharan Africa  Sao Tome and Principe           Fruits   
4                  Sub-Saharan Africa                 Rwanda  Office Supplies   
..                                ...                    ...              ...   
95                 Sub-Saharan Africa                   Mali          Clothes   
96                               Asia               Malaysia           Fruits   
97                 Sub-Saharan Africa           Sierra Leone       Vegetables   
98                      North America                 Mexico    Personal Care   
99                 Sub-Saharan Africa             Mozambique        Household   

   Sales Channel Order Priority  Order Date   Order ID   Ship Date  \
0        Offline              H   5/28/2010  669165933   6/27/2010   
1         Online              C   8/22/2012  963881480   9/15/2012   
2        Offline              L  05-02-2014  341417157  05-08-2014   
3         Online              C   6/20/2014  514321792  07-05-2014   
4        Offline              L  02-01-2013  115456712  02-06-2013   
..           ...            ...         ...        ...         ...   
95        Online              M   7/26/2011  512878119  09-03-2011   
96       Offline              L  11-11-2011  810711038  12/28/2011   
97       Offline              C  06-01-2016  728815257   6/29/2016   
98       Offline              M   7/30/2015  559427106  08-08-2015   
99       Offline              L  02-10-2012  665095412   2/15/2012   

    Units Sold  Unit Price  Unit Cost  Total Revenue  Total Cost  Total Profit  
0         9925      255.28     159.42     2533654.00  1582243.50     951410.50  
1         2804      205.70     117.11      576782.80   328376.44     248406.36  
2         1779      651.21     524.96     1158502.59   933903.84     224598.75  
3         8102        9.33       6.92       75591.66    56065.84      19525.82  
4         5062      651.21     524.96     3296425.02  2657347.52     639077.50  
..         ...         ...        ...            ...         ...           ...  
95         888      109.28      35.84       97040.64    31825.92      65214.72  
96        6267        9.33       6.92       58471.11    43367.64      15103.47  
97        1485      154.06      90.93      228779.10   135031.05      93748.05  
98        5767       81.73      56.67      471336.91   326815.89     144521.02  
99        5367      668.27     502.54     3586605.09  2697132.18     889472.91  

[100 rows x 14 columns]